### PR TITLE
Add sha1_git to file_infos #624

### PR DESCRIPTION
**Pull request by Chaitya62**
Signed-off-by: Chaitya Shah <chaitya.shah@somaiya.edu>

### CODE REVIEW

**Comment by Chaitya62** [comment]
@pombredanne  I think I covered everything, ping me if something needs to be change.

**Comment by pombredanne** [comment]
Thank you! I will merge this as soon as #885 is merged as this changes the architecture quite a bit, but the gist of the code should mergeable.

**Comment by Chaitya62** [comment]
Great, ping me here if anything needs to be changed after the merge

**Comment by pombredanne** [comment]
The merge took place... So the plugin that collects file info is the one to update. Thank you.

**Comment by pombredanne** [comment]
@Chaitya62 do you think you can update this based on the latest develop? sorry for the extra work!

**Comment by Chaitya62** [comment]
@pombredanne no problem, I'll update it.

**Comment by Chaitya62** [comment]
@pombredanne  I have updated the branch please review it

File: tests/formattedcode/data/json/tree/expected.json [hunk 1]
```diff
@@ -20,6 +20,11 @@
       "size": 91,
       "sha1": "3922760d8492eb8f853c10a627f5a73f9eaec6ff",
       "md5": "fc7f53659b7a9db8b6dff0638641778e",
+<<<<<<< HEAD
```
Comment (pombredanne) [inline tests/formattedcode/data/json/tree/expected.json]: It sounds like you committed unresolved merge conflicts?

File: src/scancode/api.py [hunk 2]
```diff
@@ -225,4 +226,4 @@ def extract_archives(location, recurse=True):
     from extractcode.extract import extract
     from extractcode import default_kinds
     for xevent in extract(location, kinds=default_kinds, recurse=recurse):
-        yield xevent
+        yield xevent
```
Comment (pombredanne) [inline src/scancode/api.py]: I think you removed the trailing line there.

File: src/scancode/api.py [hunk 3]
```diff
@@ -196,9 +196,10 @@ def get_file_info(location, **kwargs):
     result['date'] = get_last_modified_date(location) or None
     result['size'] = getsize(location) or 0
 
-    sha1, md5 = multi_checksums(location, ('sha1', 'md5',)).values()
+    sha1, md5, sha1_git = multi_checksums(location, ('sha1', 'md5','sha1_git')).values()
```
Comment (pombredanne) [inline src/scancode/api.py]: neat and clean! that's the way to go!

File: tests/formattedcode/data/json/tree/expected.json (hunk 1 above)
Comment (Chaitya62) [reply tests/formattedcode/data/json/tree/expected.json]: oh sorry, I must have missed it

**Comment by Chaitya62** [comment]
Yes, I'll take care of it this weekend

**Comment by pombredanne** [comment]
I respun the build. It may just be some CI quirk

**Comment by pombredanne** [comment]
So this is all good to go. But I have been thinking that may be this should not be part of the info after all.
May be there should be a plugin for checksums with this one and future ones (sha256, sha512, blake2, etc)... What do you think?

**Comment by Chaitya62** [comment]
Then we will have a different flag for checksums as well right? if yes then, from user point of view won't I want to have all the checksums when I ask for file infos ?

**Comment by pombredanne** [comment]
> Then we will have a different flag for checksums as well right? if yes then, from user point of view won't I want to have all the checksums when I ask for file infos ?

IMHO this would be a scanner plugin of its own. I would have a single option that could accept multiple values. Each values would be in a list of choices such as sha1-git, sha256, etc... and would add a `checksums` attribute to the scans.

something like 
```
"checksums" : [
  "sha256:12142512412121acfd",
  "sha512:12142512412121acfd",
  "sha1-git:12142512412121acfd",
]
```

**Comment by pombredanne** [comment]
@Chaitya62 any feedback on the suggestion above?

**Comment by pombredanne** [comment]
I will eventually merge this but reworked in a checksums plugin instead. Unless you care to tackle this?

**Comment by pombredanne** [comment]
@Chaitya62 Thanks! let me reopen this as a reminder under you have it in.

**Comment by majurg** [comment]
@Chaitya62 @pombredanne Ping. This has been hanging around for a while, unsure if we want to merge or not.

**Comment by Chaitya62** [comment]
After, I wrote this the code was changed significantly ( architecture wise )

**Comment by majurg** [comment]
@Chaitya62 Any plans to update this or should I close this?

**Comment by Chaitya62** [comment]
you can close it

**Comment by pombredanne** [comment]
@Chaitya62 actually I still want this... so we will start from your code to get a proper plugin in :)

**Comment by pombredanne** [comment]
@Chaitya62 I will eventually reuse bit of this and I made a copy of your branch to https://github.com/nexB/scancode-toolkit/tree/Chaitya62/add-git-sha1-to-checksums

**Comment by pombredanne** [comment]
So I am closing this for now. Thank you++

**Pull request closed without merging**